### PR TITLE
[SPARK-11485][SQL] Make DataFrameHolder and DatasetHolder public.

### PR DESCRIPTION
**Pull request by rxin**
These two classes should be public, since they are used in public code.

### CODE REVIEW

**Comment by rxin** [comment]
cc @marmbrus

File: sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala [hunk 1]
```diff
@@ -52,6 +52,10 @@ abstract class SQLImplicits {
     DatasetHolder(_sqlContext.createDataset(rdd))
   }
 
+  /**
+   * Creates a [[Dataset]] from a local Seq of Product.
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala]: This is not right, we also support Seq of Int. Maybe just say "a local Seq"

Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala]: fixed

**Comment by yhuai** [comment]
test this please

**Comment by yhuai** [comment]
LGTM

**Comment by rxin** [comment]
Merging this.

**Pull request closed without merging**